[Images] Added new Direct Creator Upload v2 (#3397)
cloudflare/cloudflare-docs: products/images/src/content/cloudflare-images/upload-images/direct-creator-upload.md

```bash
curl --request POST \

>  --url https://api.cloudflare.com/client/v4/accounts/<ACCOUNT_ID>/images/v2/direct_upload \

Reviewer: Nit: We usually don't use long parameter names for `curl`. We'll probably revisit all `curl` commands after getting our guidelines approved.

 --header 'Content-Type: application/json' \
 --header 'Authorization: Bearer :token' \

>  --form 'requireSignedURLs=true' \

Reviewer: Using cURL's [`--form` parameter](https://curl.se/docs/manpage.html#-F) with HTTP requests automatically changes the content type to `multipart/form-data`, which would make line 15 not just redundant but also incorrect. Can you check this command?

Reviewer 2: yeah, that is correct, we can remove that

 --form 'metadata={"key":"value"}'
```

You will receive a response similar to this:

```json
{

>  "result": {

Reviewer: The first-level indent should have 2 spaces, like other levels... Any reason for keeping only one space here?

Author: Actually, I'm gonna check with the eng this one

   "id": "2cdc28f0-017a-49c4-9ed7-87056c83901",
   "uploadURL": "https://upload.imagedelivery.net/2cdc28f0-017a-49c4-9ed7-87056c83901"
 },
 "result_info": null,
 "success": true,
 "errors": [],
 "messages": []
}
```

In the example above, `id` is a future image identifier that will be uploaded by a creator.

<Aside type="note" header="Note">

> Previously, with `direct_upload` v1, the ID was an identifier of a request, not an image itself. Therefore, there was no way to know if an image had been really uploaded.

Reviewer: ```suggestion
Previously, in version 1 of the Images API, the ID was an identifier of a request, not an image itself. Therefore, there was no way to know if an image had been really uploaded.
```
The version probably to the entire API and not just a single method. Can you check if my proposed change makes sense?

Also, "itself" is mostly used with definite articles (`the image itself`) and not indefinite articles (`an image itself`), which sounds a bit odd (you may want to rephrase it).

Reviewer 2: >The version probably applies to the entire Images API and not just a single method.

this applies only to a specific version of direct_upload method

</Aside>

> With `direct_upload` v2, a new draft image record is created. It will not appear on a [list of images](https://api.cloudflare.com/#cloudflare-images-list-images ), but  it is possible to fetch an image record with the provided ID to see its current status: 

Reviewer: ```suggestion
In version 2 of the Images API, a new draft image record is created when you invoke the `direct_upload` endpoint. It will not appear on a [list of images](https://api.cloudflare.com/#cloudflare-images-list-images), but  it is possible to fetch an image record with the provided ID to check its current status: 
```

Reviewer 2: >With `direct_upload` v2,

same as above, we should keep it

```bash
curl  --url https://api.cloudflare.com/client/v4/accounts/<ACCOUNT_ID>/images/v1/2cdc28f0-017a-49c4-9ed7-87056c83901 \
 --header 'Content-Type: application/json' \
 --header 'Authorization: Bearer :token'
```

You will receive a response similar to this:

```json
{

>  "result": {

Reviewer: I'd use a 2-space first-level indent.

   "id": "2cdc28f0-017a-49c4-9ed7-87056c83901",
   "metadata": {
     "key": "value":
   },
   "uploaded": "2022-01-31T16:39:28.458Z",
   "requiredSignedURLs": true,
   "variants": [
     "https://imagedelivery.net/Vi7wi5KSItxGFsWRG2Us6Q/2cdc28f0-017a-49c4-9ed7-87056c83901/public",
     "https://imagedelivery.net/Vi7wi5KSItxGFsWRG2Us6Q/2cdc28f0-017a-49c4-9ed7-87056c83901/thumbnail" 
   ],
   "draft": true
 },
 "success": true,
 "errors": [],
 "messages": []
}
```

> Once the image data is uploaded, the draft property is false and will not be in a response anymore.

Reviewer: ```suggestion
Once the image data is uploaded, the `draft` property will change to `false` and will not be part of the response anymore.
```

> Your back-end endpoint should return the `uploadURL` property to the client, enabling it to upload the image without needing to pass any authentication information with it. Here is an example of a simple HTML page that takes a one-time upload URL and uploads any selected image:

Reviewer: ```suggestion
Your back-end endpoint should return the `uploadURL` property to the client, enabling it to upload the image without needing to pass any authentication information with it. Here is an example of a simple HTML page that takes a one-time upload URL and uploads any image the user selects:
```

> If you want to override this option, add the following argument to the cURL command:

Reviewer: ```suggestion
To override this option, add the following argument to the cURL command:
```
